TST: interpolate: avoid pytest deprecations

ev-br opened the pull request:
Fix a deprecated `pytest.raises(..., message=...)` use, which somehow slipped past gh-9264.
Obsevered in gh-9661

larsoner commented:
Some of the failures are the FFT ones in `master`, but this one looks related:

https://travis-ci.org/scipy/scipy/jobs/476302856#L1260

ev-br commented:
Yeah, good catch. The exception message actually *was* different, and `raises(..., match="...")` did catch it, while `... message=...` did not :-). 
Replaced with a correct exception class, and removed the check for the message entirely. I believe what matters is the exception class, and we should be able to change the message if needs be. Hence the message itself is not IMO a part of the API.

larsoner commented:
Thanks @ev-br

tylerjereddy commented:
I think I'll have to backport this as part of https://github.com/scipy/scipy/pull/9683
